alerting/src/main/kotlin/org/opensearch/alerting/workflow/CompositeWorkflowRunner.kt

```diff
@@ -177,4 +258,102 @@ object CompositeWorkflowRunner : WorkflowRunner() {
+            for (action in trigger.actions) {
+                triggerRunResult.actionResults[action.id] = this.runAction(action, actionCtx, monitorCtx, workflow, dryRun)
+            }
+            val alert = monitorCtx.alertService!!.composeChainedAlert(
+                triggerCtx, executionId, workflow, triggerRunResult.associatedAlertIds.toList()
+            )
+            if (!dryRun && workflow.id != Workflow.NO_ID) {
+                monitorCtx.retryPolicy?.let {
+                    monitorCtx.alertService!!.saveAlerts(
+                        dataSources,
+                        listOf(alert),
+                        it,
+                        routingId = workflow.id
+                    )
+                }
+            }
+        }
+    }
+
+    private suspend fun fetchAlertsGeneratedInCurrentExecution(
+        dataSources: DataSources,
+        executionId: String,
+        monitorCtx: MonitorRunnerExecutionContext,
+        workflow: Workflow,
+    ): MutableMap<String, MutableSet<String>> {
+        try {
+            val searchRequest =
+                SearchRequest(getDelegateMonitorAlertIndex(dataSources, workflow, monitorCtx.alertIndices!!.isAlertHistoryEnabled()))
+            val queryBuilder = boolQuery()
+            queryBuilder.must(QueryBuilders.termQuery("execution_id", executionId))
+            queryBuilder.must(QueryBuilders.termQuery("state", getDelegateMonitorAlertState(workflow)))
+            val noErrorQuery = boolQuery()
+                .should(boolQuery().mustNot(existsQuery(Alert.ERROR_MESSAGE_FIELD)))
+                .should(termsQuery(Alert.ERROR_MESSAGE_FIELD, ""))
+            queryBuilder.must(noErrorQuery)
+            searchRequest.source().query(queryBuilder)
+            val searchResponse: SearchResponse = monitorCtx.client!!.suspendUntil { monitorCtx.client!!.search(searchRequest, it) }
+            val alerts = searchResponse.hits.map { hit ->
+                val xcp = XContentHelper.createParser(
+                    monitorCtx.xContentRegistry, LoggingDeprecationHandler.INSTANCE,
+                    hit.sourceRef, XContentType.JSON
+                )
+                XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.nextToken(), xcp)
+                val alert = Alert.parse(xcp, hit.id, hit.version)
+                alert
+            }
+            val map = mutableMapOf<String, MutableSet<String>>()
+            for (alert in alerts) {
+                if (map.containsKey(alert.monitorId)) {
+                    map[alert.monitorId]!!.add(alert.id)
+                } else {
+                    map[alert.monitorId] = mutableSetOf(alert.id)
+                }
+            }
+            return map
+        } catch (e: Exception) {
+            logger.error("failed to get alerts generated by delegate monitors in current execution $executionId", e)
+            return mutableMapOf()
+        }
+    }
+
+    // todo use flag in workflow
```

Comment:
Can you add more details to the todo if this will be worked on later?

Reply:
the TODO had been resolved. removed it.